src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +227,127 @@ def storage(request):
+            'size': shared_size,
+            'used': shared_used
+        },
+        'dips': {
+            'description': 'DIP uploads',
+            'path': dip_upload_dir,
+            'size': shared_size,
+            'used': dip_upload_used
+        },
+        'backlog': _usage_location_info('BL', shared_size)
+    }
+
+def _usage_check_directory_volume_size(path):
+    # Get volume size (in 512 byte blocks)
+    df = subprocess.Popen(["df", path], stdout=subprocess.PIPE)
+    output = df.communicate()[0]
+    device, size, used, available, percent, mountpoint = output.split("\n")[1].split()
+    return int(size) * 512
+
+def _usage_check_directory_used(path):
+    # Get total usage in bytes
+    du = subprocess.Popen(["du", "-b", "-c", path], stdout=subprocess.PIPE)
+
+    # Parse type total from command output
+    output = subprocess.check_output(["tail", "-n", "1"], stdin=du.stdout)
+    return output.split("\t")[0]
+
+def _usage_location_info(purpose, space_size):
+    # Get currently processing location
+    locations = storage_service.get_location(purpose=purpose)
+    location = locations[0]
+    if location['quota'] == None:
+        location_size = space_size
+    else:
+        location_size = location['quota']
+
+    # Determine purpose label
+    purpose_labels = {
+      'BL': 'Transfer Backlog'
+    }
+
+    # Calculate usage locally
+    used = _usage_check_directory_used(location['path'])
+
+    return {
+        'purpose': purpose,
+        'description': purpose_labels[purpose],
+        'path': location['path'],
+        'size': location_size,
+        'used': used}
+
+def usage_clear(request, dir_id):
+    if request.method == 'POST':
+        usage_dirs = _usage_dirs(False)
+        dir = usage_dirs[dir_id]
+
+        # Prevent shared directory from being cleared
+        if dir_id == 'shared' or not dir:
+            raise Http404
+
+        # Determine which directories need to be cleared
+        if 'purpose' in dir and dir['purpose'] == 'BL':
+            # Get location data
+            locations = storage_service.get_location(purpose=dir['purpose'])
+            location = locations[0]
+
+            dirs_to_empty = []
+            dirs_to_empty.append(os.path.join(location['path'], 'arrange'))
+            dirs_to_empty.append(os.path.join(location['path'], 'originals'))
+        else:
+            dirs_to_empty = [dir['path']]
+
+        # Clear directories
+        for dir in dirs_to_empty:
+            for entry in os.listdir(dir):
+                entry_path = os.path.join(dir, entry)
+                try:
+                    if os.path.isfile(entry_path):
+                        os.unlink(entry_path)
+                    else:
+                        shutil.rmtree(entry_path)
+                except Exception, e:
+                    messages.error(request, str(e))
+                    return redirect('components.administration.views.usage')
+
+        messages.info(request, 'Cleared.')
```

Comment:
Would be nice if this displayed the path(s) that were deleted, for better usability.